notify/notify_windows.go

```diff
@@ -34,22 +44,72 @@ func Ready() error {
 	return nil
 }
 
+// Reloading notifies the SCM that the service is entering a transitional
+// state.
 func Reloading() error {
 	if globalStatus != nil {
 		globalStatus <- svc.Status{State: svc.StartPending}
 	}
 	return nil
 }
 
+// Stopping notifies the SCM that the service is in the process of stopping.
+// This allows Windows to track the shutdown transition properly.
 func Stopping() error {
 	if globalStatus != nil {
 		globalStatus <- svc.Status{State: svc.StopPending}
 	}
 	return nil
 }
 
-// TODO: not implemented
-func Status(_ string) error { return nil }
+// Status sends an arbitrary service state to the SCM based on a string
+// identifier. This allows higher-level code to map custom states to
+// Windows service states when appropriate.
+func Status(name string) error {
+	if globalStatus == nil {
+		return nil
+	}
+
+	var state svc.State
+	var accepts svc.Accepted
+	accepts = 0
+
+	switch strings.ToLower(name) {
```

Comment:
On Linux, Status accepts any descriptive message.
This PR makes the Windows version fail for any unknown string, which creates unexpected cross-platform differences.
Unknown status names should be ignored (or logged) rather than returned as errors, and comments should clarify that Windows only supports a limited set of predefined states.

BTW, I think we could ask @mholt for his opinion and see if he has any other plans for Status.